rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -218,84 +220,83 @@ def required_packages(cls) -> List[Text]:
 
     # training data helpers:
     @staticmethod
-    def _create_label_id_dict(training_data: "TrainingData") -> Dict[Text, int]:
+    def _create_label_id_dict(
+        training_data: "TrainingData", attribute: Text
+    ) -> Dict[Text, int]:
         """Create label_id dictionary"""
 
         distinct_label_ids = set(
-            [example.get("intent") for example in training_data.intent_examples]
-        )
+            [example.get(attribute) for example in training_data.intent_examples]
+        ) - {None}
```

Comment:
where is that `None` coming from?

Reply:
The function is shared between embedding intent classifier and response selector(child class). `training_data.intent_examples` can contain training examples where `response` attribute is set to `None`. So either I skip those examples by using a `if example.get(attribute)` or remove `None` from the set.